bark/world/world.cpp

```diff
@@ -61,17 +61,20 @@ void World::PlanAgents(const float& delta_time) {
   WorldPtr current_world(this->Clone());
   const float inc_world_time = world_time_ + delta_time;
   for (auto agent : agents_) {
-    ObservedWorld observed_world(current_world, agent.first);
-    agent.second->PlanBehavior(delta_time, observed_world);
-    if (agent.second->GetBehaviorStatus() == BehaviorStatus::VALID)
-      agent.second->PlanExecution(inc_world_time);
+    if (agent.second->IsValidAtTime(world_time_)) {
```

Comment:
who sets if it is valid?

Reply:
Agent has a function 
`bool Agent::IsValidAtTime(const float& world_time) const {
  return isgreaterequal(world_time + std::numeric_limits<float>::epsilon(),
                        first_valid_timestamp_);
}
`

Reply:
default is first_valid_timestamp_ = 0.0
when simulating the dataset in closed loop, these agents get different first_valid_timestamps.
everything else should work as usual